Client/Frontend/Settings/FxAContentViewController.swift

```diff
@@ -164,6 +172,24 @@ class FxAContentViewController: SettingsContentViewController, WKScriptMessageHa
             handleRemoteCommand(detail["command"].stringValue, data: detail["data"])
         }
     }
+    
+    // Configure the FxA signin url based on any passed options.
+    func FxAURLWithOptions(_ fxaOptions: FxALaunchParams, profile: Profile) -> URL {
+        // Append any passed query strings to the signin url. Note that you can't
+        // override the service and context params.
+        guard fxaOptions.query != nil else {
+            return profile.accountConfiguration.signInURL
+        }
+        
+        var queryParams: String = ""
+        for param in fxaOptions.query! {
```

Comment:
You should be able to build params using query["service"] rather than using a for loop